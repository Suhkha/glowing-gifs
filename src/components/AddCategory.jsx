import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({target}) => {
        setInputValue(target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        const cleanValue = inputValue.trim();
        if(cleanValue.length <= 1) return;

        // setCategories( categories => [...categories, inputValue]);
        onNewCategory(cleanValue);
        setInputValue('');
    }

    return (
        <form onSubmit={(event) => onSubmit(event) } aria-label="form">
            <input 
                type="text" 
                placeholder="Search gifs" 
                value={ inputValue }
                onChange={ onInputChange }
            />
        </form>
        
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}