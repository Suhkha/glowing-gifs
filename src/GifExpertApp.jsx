import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['BTS']);

  const onAddCategory = (newCategory) => {
    if(categories.includes(newCategory)) return;

    //setCategories([...categories, 'Taylor Swift']); // Opcion 1
    setCategories( cat => [newCategory, ...categories]); // Opcion 2
  }

  return (
    <>
        <h1>GiftExpertApp</h1>
        
        <AddCategory onNewCategory={ onAddCategory } />

        { 
          categories.map((category) => (
            <GifGrid 
              key={ category } 
              category={ category } 
            />
          ))
        }
    </>
  )
}
