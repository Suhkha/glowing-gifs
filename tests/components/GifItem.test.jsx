import {render, screen} from "@testing-library/react";
import { GifItem } from "../../src/components";

describe('Testing GifItem', () => { 
    const title = "Main title";
    const url = "https://google.com";

    test('it should match with the snapshot', () => { 

        const { container } = render(<GifItem title={ title } url={ url }/>)
        expect(container).toMatchSnapshot();
    });

    test('show image with its URL and ALT', () => {
        render(<GifItem title={ title } url={ url }/>)
        //screen.debug();
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);

        const {src, alt} = screen.getByRole('img');
        expect(src).toMatch(url);
        expect(alt).toMatch(title)
    });

    test('show title in the component', () => {
        render(<GifItem title={ title } url={ url }/>)
        expect(screen.getByText(title)).toBeTruthy();
    })
})