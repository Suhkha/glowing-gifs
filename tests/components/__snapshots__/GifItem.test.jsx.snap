// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifItem it should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Main title"
      src="https://google.com"
    />
    <p>
      Main title
    </p>
  </div>
</div>
`;

exports[`Testing GifItem it should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Main title"
      src="https://google.com"
    />
    <p>
      Main title
    </p>
  </div>
</div>
`;
